utils/hedge.go

```diff
@@ -0,0 +1,57 @@
+package utils
+
+import (
+	"context"
+	"time"
+
+	"go.uber.org/multierr"
+)
+
+type HedgeParams[T any] struct {
+	Timeout       time.Duration
+	RetryDelay    time.Duration
+	MaxAttempts   int
+	IsRecoverable func(err error) bool
+	Func          func(context.Context) (T, error)
+}
+
+// race retries if the function takes to long to return
+// |---------------- attempt 1 ----------------|
+// |    delay    |--------- attempt 2 ---------|
+func HedgeCall[T any](ctx context.Context, params HedgeParams[T]) (v T, err error) {
```

Comment:
There is a named return for `err` and use of `err` inside also. I think it is fine. Just wanted to flag it because we got bitten by it recently :-)

Reply:
`err` accumulates errors from `res.err` and if the call fails it returns all of the errors that happened during the call. but if we reach the return condition we only want to return the value and error from res

Reply:
oh the `err` in the closure. no that won't affect the err from the return